repose-aggregator/components/services/akka-http-client-service/akka-http-client-service-impl/src/test/scala/org/openrepose/core/services/serviceclient/akka/impl/AkkaServiceClientImplReflectionTest.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.core.services.serviceclient.akka.impl
+
+import javax.ws.rs.core.MediaType
+
+import org.apache.commons.lang3.reflect.FieldUtils
+import org.apache.http._
+import org.apache.http.client.HttpClient
+import org.apache.http.client.methods.HttpUriRequest
+import org.apache.http.message.BasicHttpResponse
+import org.apache.http.params.HttpParams
+import org.mockito.Mockito.{times, verify, when}
+import org.mockito.{Matchers => MockitoMatchers}
+import org.openrepose.commons.utils.http.ServiceClientResponse
+import org.openrepose.core.service.httpclient.config.{HttpConnectionPoolConfig, PoolType}
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.core.services.httpclient.{HttpClientContainer, HttpClientService}
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach, FunSpec}
+
+import scala.collection.JavaConversions._
+
+// TODO: Remove this once the cache TTL (and other necessary fields) can be set by other means
+class AkkaServiceClientImplReflectionTest extends FunSpec with BeforeAndAfterAll with BeforeAndAfterEach with MockitoSugar {
```

Comment:
Well, `AkkaServiceClientImplTest` already exists, but I wanted to break this into its own test for a couple of reasons:

1. I don't love using reflection. This makes it clear that reflection is being used, and that once it is not longer needed, should not be used. Also, I'm not certain how the test runner runs tests, but I hoped that putting this in its own test class would prevent other tests from seeing the changes made to the class through reflection. I actually bet other tests do see the reflective change, but it shouldn't overly matter as none of our unit tests require that the timeout be 500ms.
2. The other test is kind of messy.